ts/src/okx.ts

```diff
@@ -6736,6 +6737,44 @@ export default class okx extends Exchange {
         return result;
     }
 
+    async fetchUnderlyingAssets (marketType: string = undefined, params = {}) {
+        /**
+         * @method
+         * @name okx#fetchUnderlyingAssets
+         * @description fetches the market ids of underlying assets for a specific contract market type
+         * @see https://www.okx.com/docs-v5/en/#public-data-rest-api-get-underlying
+         * @param {string|undefined} marketType the contract market type, 'option', 'swap' or 'future', the default is 'option'
+         * @param {object} [params] exchange specific params
+         * @returns {object[]} a list of [underlying assets]{@link https://github.com/ccxt/ccxt/wiki/Manual#underlying-assets-structure}
+         */
+        await this.loadMarkets ();
+        [ marketType, params ] = this.handleMarketTypeAndParams ('fetchUnderlyingAssets', undefined, params);
```

Comment:
@Dan-krm this is confusing by doing this you're overriding the `marketType` argument, therefore ignoring it

Reply:
Restored the previous signature it is simpler